assets/tests/hooks/useDetour.test.ts

```diff
@@ -519,7 +541,26 @@ describe("useDetour", () => {
         )
       )
 
-      expect(result.current.stops).toBe(stops)
+      expect(result.current.stops).toStrictEqual(stops)
+    })
+
+    test("`stops` is de-duplicated", () => {
+      const stop1 = stopFactory.build()
+      const stop2 = stopFactory.build()
+      const stop3 = stopFactory.build()
+      const stop4 = stopFactory.build()
+
+      const { result } = renderHook(() =>
+        useDetour(
+          originalRouteFactory.build({
+            shape: shapeFactory.build({
+              stops: [stop1, stop2, stop3, stop2, stop4],
+            }),
+          })
+        )
+      )
+
+      expect(result.current.stops).toStrictEqual([stop1, stop2, stop3, stop4])
```

Comment:
Hmmmm...

So if we don't de-dup, then the effect will be that we draw icons over each other. I can't imagine that that's the behavior we want.

Also, and admittedly this is easily solvable, but React yells at us if there are duplicates, because the `key` value in the array under `ZoomLevelWrapper` is duplicated.

Reply:
> Also, and admittedly this is easily solvable, but React yells at us if there are duplicates, because the `key` value in the array under `ZoomLevelWrapper` is duplicated.

My opinion is that we're doing something wrong by treating `stop_id` as the primary key here.

> So if we don't de-dup, then the effect will be that we draw icons over each other. I can't imagine that that's the behavior we want.

Yeah, I get why we're doing it, and I know we don't have enough information to actually do anything about this right now, so again, this is non-blocking, but I wanted to point this out because this now not only applies to the panel text, but also the map[though I do acknowledge that the map would've failed on duplicated stops anyway, because we're using `stop_id` as the primary key].